ReactiveXComponent/Connection/IXCPublisher.cs

```diff
@@ -0,0 +1,11 @@
+using System;
+using ReactiveXComponent.Common;
+
+namespace ReactiveXComponent.Connection
+{
+    public interface IXCPublisher : IDisposable
+    {
+        void InitPrivateCommunication(string privateCommunicationIdentifier);
+        void SendEvent(string engine, string component, string stateMachine, int eventCode, string messageType, object message, Visibility visibility);
```

Comment:
too many parameters !!!!
think as a user, it's a public interface
just sendEvent(component, stateMachine, message, Visibility?)